bounties/MetaMask Snap for the XRP Ledger.md

```diff
@@ -0,0 +1,38 @@
+---
+ID: "0042"
+Discussion: https://github.com/XRPLBounties/Proposals/discussions/0042
+Category: Browser Extension
+Status: Open
+Amount: $100,000
+---
+
+# MetaMask Snap for the XRP Ledger
+
+## Overview
+
+[MetaMask](https://metamask.io/) is a gateway to blockchain apps with over 30 million users. With [Snaps](https://metamask.io/snaps/), developers can extend MetaMask to add additional features by leveraging existing APIs, such as the [XRP Ledger API](https://xrpl.org/public-api-methods.html). The Snap may also leverage JavaScript libraries such as [xrpl.js](https://github.com/XRPLF/xrpl.js) for functions like client-side key generation and transaction signing.
+
+## Details
+
+Develop, publish, and support an open source application allowing users to directly manage XRP and XRPL tokens within the MetaMask interface. The Snap should allow users to interact with XRP and XRPL tokens (on the XRP Ledger) without having to wrap tokens on Ethereum or any other blockchain.
+
+Snaps is a system that allows developers to build on the capabilities of MetaMask. It is a program that can customize and modify MetaMask's experience for end users. This Snap must add new APIs to MetaMask and add support for the XRP Ledger blockchain. See the [MetaMask Docs about Snaps for more info](https://docs.metamask.io/guide/snaps.html).
+
+As an example, the Snap would be similar to [BitcoinSnap](https://github.com/KeystoneHQ/btcsnap), but for the XRP Ledger.
+
+## Milestones
+
+| # | Description | Details | Proposed Potential Award |
+| - | ----------- | ------- | ------------------------ |
+| 1 | Create Account | Implement a Snap that securely generates and stores a new keypair and displays the associated XRPL address to the user. Make it easy for users to copy the address to share or use in other apps. | $1,000 |
+| 2 | Send XRP | Enable users to send XRP. The Snap should sign and submit the transaction to a server of the user's choice. For convenience, users can select from one of the [well-known public servers](https://xrpl.org/public-servers.html). | $2,000 |
+| 3 | View Transaction History | Show transaction history in a human-readable UI. | $10,000 |
+| 4 | Set Trust Line | Enable users to [create or modify a trust line](https://xrpl.org/trustset.html). | $2,000 |
+| 5 | Sign Anything | Enable users to sign any transaction or message. | $3,000 |
+| 6 | Sign and Submit Transaction | Enable users to sign and submit any transaction. | $3,000 |
+| 7 | Dapp Integration | Offer APIs for dapps to interact with the Snap. For an example, see the [API to integrate BitcoinSnap into your dapp](https://github.com/KeystoneHQ/btcsnap/tree/master/packages/snap). | $4,000 |
```

Comment:
IMO this should be worth more, if API documentation is included

Reply:
```suggestion
| 7 | Dapp Integration | Offer APIs for dapps to interact with the Snap, with API documentation and snippets of example code. As a reference, see the [API to integrate BitcoinSnap into your dapp](https://github.com/KeystoneHQ/btcsnap/tree/master/packages/snap). | $8,000 |
```